iot_config/src/region_map.rs

```diff
@@ -76,19 +68,28 @@ pub async fn build_region_tree(
     while let Some(region_row) = regions.try_next().await? {
         let region = Region::from_i32(region_row.region)
             .ok_or(RegionMapError::UnsupportedRegion(region_row.region))?;
-        let mut idx_decoder = Decoder::new(&region_row.indexes[..])?;
-        let mut buf = Vec::new();
-        idx_decoder.read_to_end(&mut buf)?;
-
-        let mut idx_buf = [0_u8; 8];
-        for chunk in buf.chunks(8) {
-            if chunk.len() < idx_buf.len() {
```

Comment:
note that I moved this check out of the hot-loop by doing a single is-multiple-of-8 check before